peoplehr/client.bal

```diff
@@ -0,0 +1,260 @@
+        http:Request request = new;
+        json jsonBody = check payload.cloneWithType(json);
+        json authentication =  {APIKey: self.config.apiKey, Action: "GetSalaryDetail"};
+        request.setJsonPayload(check jsonBody.mergeJson(authentication));
+        return self.basicClient->post(SALARY, request, targetType = GetSalaryDetailResponse);      
+    }
+
+    # Add new holiday.
+    #
+    # + payload - Request detail on new holiday
+    # + return - Successful response or an error
+    @display{label: "Add New Holiday"}
+    isolated remote function addNewHoliday(@display{label: "Holiday Detail"} NewHolidayRequest|json payload) returns 
+                                            AddNewHolidayResponse|error {
+        http:Request request = new;
+        json jsonBody = check payload.cloneWithType(json);
+        json authentication =  {APIKey: self.config.apiKey, Action: "AddNewHoliday"};
+        request.setJsonPayload(check jsonBody.mergeJson(authentication)); 
+        return self.basicClient->post(HOLIDAY, request, targetType = AddNewHolidayResponse);       
+    }
+
+    # Gets holiday detail list
+    #
+    # + payload - Request detail on a holiday
+    # + return - Successful response or an error
+    @display{label: "Get Holiday Detail"}
+    isolated remote function getHolidayDetail(@display{label: "Holiday Detail"} HolidayDetail payload) returns 
+                                              GetHolidayDetailResponse|error {
+        http:Request request = new;
+        json jsonBody = check payload.cloneWithType(json);
+        json authentication =  {APIKey: self.config.apiKey, Action: "GetHolidayDetail"};
+        request.setJsonPayload(check jsonBody.mergeJson(authentication)); 
+        return self.basicClient->post(HOLIDAY, request, targetType = GetHolidayDetailResponse);      
+    }
+
+    # Deletes holiday detail
+    #
+    # + payload - Request detail on a holiday
+    # + return - Successful response or an error
+    @display{label: "Delete Holiday"}
+    isolated remote function deleteHoliday(@display{label: "Holiday Detail"} HolidayDetail payload) returns 
+                                            DeleteHolidayResponse|error {
+        http:Request request = new;
+        json jsonBody = check payload.cloneWithType(json);
+        json authentication =  {APIKey: self.config.apiKey, Action: "DeleteHoliday"};
+        request.setJsonPayload(check jsonBody.mergeJson(authentication)); 
+        return self.basicClient->post(HOLIDAY, request, targetType = DeleteHolidayResponse);      
+    }
+
+    # Gets query result By query name details
+    #
+    # + payload - Request detail on a query 
+    # + return - Successful response or an error
+    @display{label: "Get Query Result By Query Name"}
+    isolated remote function getQueryResultByQueryName(@display{label: "Request Detail on Query Result"} 
```

Comment:
I am not clean what this method does. Is it `getQueryByName`? PeopleHR users would know there will be something called queries.